src/components/HideUserButton/BlockUserButton.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button, ButtonProps, Menu } from '@mantine/core';
+import { openConfirmModal } from '@mantine/modals';
+import { IconUserCancel, IconUserCheck } from '@tabler/icons-react';
+import { MouseEventHandler } from 'react';
+import { LoginRedirect } from '~/components/LoginRedirect/LoginRedirect';
+import { useHiddenPreferencesData, useToggleHiddenPreferences } from '~/hooks/hidden-preferences';
+
+import { useCurrentUser } from '~/hooks/useCurrentUser';
+import { showSuccessNotification } from '~/utils/notifications';
+
+export function BlockUserButton({ userId, as = 'button', onToggleHide, ...props }: Props) {
+  const currentUser = useCurrentUser();
+
+  const users = useHiddenPreferencesData().blockedUsers;
+  const alreadyBlocking = users.some((x) => x.id === userId);
```

Comment:
Please let's do `isBlocked` xD no way I'm gonna think to search for `already...`